src/core/storage/compression/TestCompressorDecompressor.cpp

```diff
@@ -3,149 +3,169 @@
-                        [ ](auto sz, auto entry) { return is_directory(entry) ? sz : sz + file_size(entry); });
-        else
-                initial_size = fs::file_size(initial_file_folder);
-        
-        unsigned long long int compressed_size = fs::file_size(compressed_file);
-        
-        REQUIRE(compressed_size < initial_size);
-        std::cout << std::endl << std::endl << "#############################################################" << std::endl;
-        std::cout << "Passed with initial size: " << initial_size << " and compressed size: " << compressed_size << std::endl;
-        std::cout << "#############################################################" << std::endl << std::endl << std::endl;
+void check_initial_compressed_size(const std::string &initial_file_folder, const std::string &compressed_file) {
+	unsigned long initial_size;
+	std::cout << "directory: " << fs::is_directory(initial_file_folder) << std::endl;
+	if (fs::is_directory(initial_file_folder))
+		initial_size = std::accumulate(
+			fs::recursive_directory_iterator(initial_file_folder.c_str()),
+			fs::recursive_directory_iterator(), 0,
+			[](auto sz, auto entry) { return is_directory(entry) ? sz : sz + file_size(entry); });
+	else
+		initial_size = fs::file_size(initial_file_folder);
+
+	unsigned long long int compressed_size = fs::file_size(compressed_file);
+
+	REQUIRE(compressed_size < initial_size);
+	std::cout << std::endl << std::endl << "#############################################################" << std::endl;
+	std::cout << "Passed with initial size: " << initial_size << " and compressed size: " << compressed_size
+	          << std::endl;
+	std::cout << "#############################################################" << std::endl << std::endl << std::endl;
 }
 
-void folder_test(std::map<std::string, std::string> &params, const int text_size)
-{
-        for (const auto &item: params)
-                REQUIRE(!exists(item.second));
-        
-        REQUIRE(create_testing_directory(params["test_dir_name"], text_size));
-        REQUIRE(exists(params["test_dir_name"]));
-        
-        compression::Compressor compress{
-                std::vector<std::string>(1, params["test_dir_name"]),
-                params["compressed_name"]
-        };
-        compress();
-        REQUIRE(exists(params["compressed_name"]));
-        
-        
-        fs::rename(params["test_dir_name"], params["changed_to_initial_dir"]);
-        REQUIRE(exists(params["changed_to_initial_dir"]));
-        
-        decompression::Decompressor decompress{params["compressed_name"]};
-        decompress();
-        REQUIRE(exists(params["test_dir_name"]));
-        
-        REQUIRE(compare_folders("InitialDir", "ForTesting"));
-        
-        check_initial_compressed_size(params["test_dir_name"], params["compressed_name"]);
-        REQUIRE(clean(params));
+void folder_test(std::map<std::string, std::string> &params, const int text_size) {
```

Comment:
Could you please split this PR into multiple commits? Maybe one for the feature impl and another one for the added tests? Feel free to add even more.